playbook.yml

```diff
@@ -1,15 +1,59 @@
 ---
 - hosts: all
+  gather_facts: False
 
   pre_tasks:
+
+    # Check Ubuntu release version to determine if we need to install python 2,
+    # an ansible dependency that isn't included by default in Ubuntu 16.04 and
+    # up.
+
+    - raw: cat /etc/lsb-release | grep DISTRIB_RELEASE | cut -d "=" -f2
+      register: ubuntu_release
+
+    - debug: msg="Running ubuntu version {{ ubuntu_release.stdout|float }}"
+
+  # Update apt cache and install python 2 for Ubuntu versions greater than 
+  # 16.04
+
     - name: Update APT cache
-      apt: update_cache=yes
+      raw: apt-get update
```

Comment:
This task has no `name`, so I think either this one is never getting applied, or only the one above it is.